tfhe/src/high_level_api/compact_list.rs

```diff
@@ -18,6 +18,57 @@ use crate::shortint::MessageModulus;
 use crate::zk::{CompactPkePublicParams, ZkComputeLoad};
 use crate::CompactPublicKey;
 
+impl crate::FheTypes {
+    fn from_data_kind(data_kind: DataKind, message_modulus: MessageModulus) -> Option<Self> {
+        Some(match data_kind {
+            DataKind::Unsigned(n) => {
+                let num_bits_per_block = message_modulus.0.ilog2() as usize;
+                let num_bits = n * num_bits_per_block;
+                match num_bits {
+                    2 => Self::Uint2,
+                    4 => Self::Uint4,
+                    6 => Self::Uint6,
+                    8 => Self::Uint8,
+                    10 => Self::Uint10,
+                    12 => Self::Uint12,
+                    14 => Self::Uint14,
+                    16 => Self::Uint16,
+                    32 => Self::Uint32,
+                    64 => Self::Uint64,
+                    128 => Self::Uint128,
+                    160 => Self::Uint160,
+                    256 => Self::Uint256,
+                    512 => Self::Uint512,
+                    1024 => Self::Uint1024,
+                    2048 => Self::Uint2048,
+                    _ => return None,
+                }
+            }
+            DataKind::Signed(n) => {
+                let num_bits_per_block = message_modulus.0.ilog2() as usize;
+                let num_bits = n * num_bits_per_block;
+                match num_bits {
+                    2 => Self::Int2,
+                    4 => Self::Int4,
+                    6 => Self::Int6,
+                    8 => Self::Int8,
+                    10 => Self::Int10,
+                    12 => Self::Int12,
+                    14 => Self::Int14,
+                    16 => Self::Int16,
+                    32 => Self::Int32,
+                    64 => Self::Int64,
+                    128 => Self::Int128,
+                    160 => Self::Int160,
+                    256 => Self::Int256,
```

Comment:
no larger signed values I'm guessing ?

Reply:
Yep, 512, 1024, and 2048 are available as uint as the blockchain requested them, but its not needed for signed ints